test/unit/menu/ExportIEDParams.test.ts

```diff
@@ -0,0 +1,424 @@
+
+    doc = await fetch('/test/testfiles/menu/export-ied-params.scd')
+      .then(response => response.text())
+      .then(str => new DOMParser().parseFromString(str, 'application/xml'));
+
+    plugin.doc = doc;
+  });
+
+  describe('determine selector (getSelector) -', () => {
+    it('when no variable in string then same string returned', () => {
+      const selector = 'LN[lnClass="TCTR"]';
+      expect(plugin['getSelector'](selector, 'IED1')).to.be.equal(selector);
+    });
+
+    it('when variable in string then variable is replaced', () => {
+      const iedName = 'IED1';
+      const expectedSelector = `ConnectedAP[iedName="${iedName}"]`;
+
+      const selectorVariant1 = 'ConnectedAP[iedName="{{ iedName }}"]';
+      expect(plugin['getSelector'](selectorVariant1, 'IED1')).to.be.equal(
+        expectedSelector
+      );
+
+      const selectorVariant2 = 'ConnectedAP[iedName="{{iedName}}"]';
+      expect(plugin['getSelector'](selectorVariant2, 'IED1')).to.be.equal(
+        expectedSelector
+      );
+
+      const selectorVariant3 = 'ConnectedAP[iedName="{{   iedName   }}"]';
+      expect(plugin['getSelector'](selectorVariant3, 'IED1')).to.be.equal(
+        expectedSelector
+      );
+    });
+  });
+
+  describe('retrieve the data template element from a type element (getDataElement) -', () => {
+    it('when called with a LNodeType element and a known name is passed then correct DO Element returned', () => {
+      const typeElement = doc.querySelector('LNodeType[id="Dummy.TVTR"]');
+
+      const dataElement = plugin['getDataElement'](typeElement!, 'Beh');
+      expect(dataElement).to.be.not.null;
+      expect(dataElement?.tagName).to.be.equal('DO');
+      expect(dataElement).to.have.attribute('type', 'OpenSCD_ENS_Beh');
+    });
+
+    it('when called with a LNodeType element and a unknown name is passed then null returned', () => {
+      const typeElement = doc.querySelector('LNodeType[id="Dummy.TVTR"]');
+
+      const dataElement = plugin['getDataElement'](typeElement!, 'Unknown');
+      expect(dataElement).to.be.null;
+    });
+
+    it('when called with a DOType element and a known name is passed then correct DO Element returned', () => {
+      const typeElement = doc.querySelector('DOType[id="Dummy.ASG"]');
+
+      const dataElement = plugin['getDataElement'](typeElement!, 'q');
+      expect(dataElement).to.be.not.null;
+      expect(dataElement?.tagName).to.be.equal('DA');
+      expect(dataElement).to.have.attribute('bType', 'Quality');
+    });
+
+    it('when called with a DOType element and a unknown name is passed then null returned', () => {
+      const typeElement = doc.querySelector('DOType[id="Dummy.ASG"]');
+
+      const dataElement = plugin['getDataElement'](typeElement!, 'Unknown');
+      expect(dataElement).to.be.null;
+    });
+
+    it('when called with a DAType element and a known name is passed then correct DO Element returned', () => {
```

Comment:
This should be `DA` or `BDA` isn't it?

Reply:
I mean instead of `DO`

Reply:
Correct, but a DAType can only have BDAs. The test was correct, only the description luckily.
Fixed it.